csv_export.php

```diff
@@ -100,29 +102,74 @@
 
 echo $t_header;
 
-# export the rows
-foreach ( $t_rows as $t_row ) {
-	$t_first_column = true;
 
-	foreach ( $t_columns as $t_column ) {
-		if( !$t_first_column ) {
-			echo $t_sep;
-		} else {
-			$t_first_column = false;
+define( 'EXPORT_BLOCK_SIZE', 500 );
+
+$t_end_of_results = false;
+do {
+	# Clear cache for next block
+	unset( $g_cache_bug );
```

Comment:
I would call the proper api to reset its own cache.  For example, bug_cache_reset().  Do the same with other purging of caches.